brave/brave-core
# Add configurable default tip amounts

## Submitter Checklist:
Resolves https://github.com/brave/brave-browser/issues/6011

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (unit, browser, security tests) on

## Test Plan:

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a verified publisher with publisher-defined default tip amounts
4. Open tip panel and choose "Send a tip..."
5. Ensure that publisher's defaults are shown in the site banner

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher *without* publisher-defined default tip amounts
4. Open tip panel and choose "Send a tip..."
5. Ensure that wallet's defaults of [1, 10, 50] BAT are shown in the site banner

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher with publisher-defined default tip amounts
4. Open tip panel and choose "Set..." to set a monthly contribution
5. Ensure that publisher's defaults are shown in the site banner

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher *without* publisher-defined default tip amounts
4. Open tip panel and choose "Set..." to set a monthly contribution
5. Ensure that wallet's defaults of [1, 10, 50] BAT are shown in the site banner

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher with publisher-defined default tip amounts
4. Open tip panel and choose "Set..." to set a monthly contribution
5. Set monthly contribution to any value and click "Set Contribution"
6. Open tip panel again
7. Ensure that "Monthly Contribution" shows the value from step 4
8. Click the value to change it
9. Ensure that publisher's defaults are shown in the menu

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher *without* publisher-defined default tip amounts
4. Open tip panel and choose "Set..." to set a monthly contribution
5. Set contribution to 1.0 BAT and click "Set Contribution"
6. Open tip panel again
7. Ensure that "Monthly Contribution" shows "1.0 BAT"
8. Click the value to change it
9. Ensure that wallet's defaults of [1, 10, 50] BAT are shown in the menu

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher with publisher-defined default tip amounts
4. Open tip panel and choose "Set..." to set a monthly contribution
5. Set monthly contribution to any value and click "Set Contribution"
6. Close the browser tab
7. Modify values in `server_publisher_amounts` table in `publisher_info_db` to change the value set in step 4 to another value. For example, if the publisher has defaults of [5, 10, 20], and you selected 5 BAT in step 4, change the value in the table from 5 to 6 BAT
8. Open a new tab with the same publisher
9. Open tip panel again
10. Ensure that "Monthly Contribution" shows the value from step 3
11. Click the value to change it
12. Ensure that publisher's *new* defaults are shown in the menu (e.g. [6, 10, 20])

### Test case
1. Launch on staging
2. Enable Rewards and claim grant
3. Visit a publisher
4. Disconnect your internet connection
5. Open the tip panel and choose "Send a tip..."
6. Ensure that values in site banner are the fallback values of [1, 5, 10]
7. Close tip panel
8. Reconnect internet connection
9. Open tip panel and choose "Send a tip..."
10. Ensure that wallet's defaults of [1, 10, 50] BAT are shown in the site banner


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on.
